### DISCUSSION TITLE
`Object self` is disabled

### DISCUSSION THREAD

**Discussion by iacore** [stevedekorte/io#469] [Q&A] [answered]
In [language reference](https://iolanguage.org/reference/index.html), `self` is mentioned.

```io

super(aMessage)
Sends the message aMessage to the receiver's proto with the context of self. Example:

	self test(1, 2)   // performs test(1, 2) on self
	super(test(1, 2)) // performs test(1, 2) on self proto but with the context of self
	

```

However, it is disabled in source code. What's the reason behind this?

**Comment by zephyrtronium** [comment, marked as answer]
`self` is set while activating blocks, along with a couple other slots: https://github.com/IoLanguage/io/blob/8eb12006c6ff4ae749ad1b5f40e862a611dfd1fd/libs/iovm/source/IoBlock.c#L226-L232

There isn't a global `self`.

**Reply by iacore** [reply]
This behavior is kind of weird

**Comment by stevedekorte** [comment]
@iacore does @zephyrtronium’s response answer your question?

**Reply by iacore** [reply]
yes

**Comment by stevedekorte** [comment]
IIRC, there may also be a thisContext or thisObject method to return the current local context.
